src/main/scala/query/MatchCreateQuery.scala

```diff
@@ -0,0 +1,74 @@
+package com.originate.scalypher
+
+import action.ReturnAction
+import action.ReturnAll
+import action.ReturnDistinct
+import action.ReturnReference
+import path.AnyNode
+import path.AnyRelationship
+import path.Node
+import path.Path
+import path.Relationship
+import where.Reference
+import types._
+
+trait MatchCreateQuery extends Query {
```

Comment:
this holds some logic since there is some special stuff that happens for creating/merging queries (a merge query is basically a "find or create")